src/nupic/regions/ScalarSensor.hpp

```diff
@@ -70,7 +70,18 @@ class ScalarSensor : public RegionImpl {
   getNodeOutputElementCount(const std::string &outputName) override;
 
 private:
-  Real64 sensedValue_;
+  struct {
+    Real64 sensedValue_;
+    Real64 resolution;
+    Real64 radius;
+    Real64 minValue;
+    Real64 maxValue;
+    UInt32 n;
+    UInt32 w;
+    bool periodic;
+    bool clipInput;
+  } params_;
```

Comment:
Very nice!  This is good organization, and C++ even saves/loads the struct for you.

Reply:
for a discussion in a new issue, with these params, I had this idea I'd like to implement eventually: 
- each class implements (has a ) struct params_ member instead of its private/public fiends. 
- makes serialization easier
- has constructor that takes JSON
- we could remove (from API) all the public getters/setters for individual values, instead use
  - `get("columnSize"), set("isGlobal", false)`